src/scripts/scenes/mainScene.ts

```diff
@@ -69,53 +72,91 @@ export default class MainScene extends Phaser.Scene {
     this.ennemis.push(new Enemy(this, width * 3, 350, 'octopus', this.submarine, this.onCollision.bind(this)))
-    this.ennemis.push(new Enemy(this, width * 4, 100, 'shark', this.submarine, this.onCollision.bind(this)))
+    this.ennemis.push(new Enemy(this, width * 3, 400, 'fish', this.submarine, this.onCollision.bind(this)))
     this.ennemis.push(new Enemy(this, width * 4, 350, 'octopus', this.submarine, this.onCollision.bind(this)))
     this.ennemis.push(new Enemy(this, width * 5, 250, 'octopus', this.submarine, this.onCollision.bind(this)))
 
-    this.fpsText = new FpsText(this)
+    this.ennemis.push(new Enemy(this, width, 200, 'shark', this.submarine, this.onCollision.bind(this)))
+    this.ennemis.push(new Enemy(this, width * 3, 350, 'shark', this.submarine, this.onCollision.bind(this)))
+    this.ennemis.push(new Enemy(this, width * 4, 400, 'shark', this.submarine, this.onCollision.bind(this)))
+
+
+    this.ennemis.push(new Enemy(this, width * 5, 200, 'fish', this.submarine, this.onCollision.bind(this)))
+    this.ennemis.push(new Enemy(this, width * 5, 300, 'fish', this.submarine, this.onCollision.bind(this)))
+
+    this.matter.world.on("collisionstart", (event, bodyA: MatterJS.BodyType, bodyB: MatterJS.BodyType) => {
+      if (bodyA.label === "submarine-light" && bodyB.label === "ennemy" && this.submarine.light.currentBody.render.visible) {
+        bodyB.gameObject.escape()
+      } else if (bodyB.label === "submarine-light" && bodyA.label === "ennemy" && this.submarine.light.currentBody.render.visible) {
+        bodyA.gameObject.escape()
+      } else if (bodyA.label === "submarine" && bodyB.label === "ennemy") {
+        this.onCollision()
+      } else if (bodyB.label === "submarine" && bodyA.label === "ennemy") {
+        this.onCollision()
+      }
+    });
+
+
+    // this.fpsText = new FpsText(this)
+
+    // this.fpsText.setPosition(0, 80).setDepth(7)
 
-    this.fpsText.setPosition(0, 80).setDepth(7)
-    
     this.matter.world.setBounds(0, 0, 2400 * NB_BACKGROUND, 600, 64, true, true, true, true)
     this.physics.world.setBounds(0, 0, width, 600, true, true, true, true)
     this.lights.enable().setAmbientColor(0x555555)
 
     this.input.keyboard.on('keydown', event => this.dealWithKeyDown(event))
     this.input.keyboard.on('keyup', event => this.resetCommand(event))
+
+    this.scoreText.setScrollFactor(0)
+    //this.fpsText.setScrollFactor(0)
+
+    this.lastScorePos = this.submarine.x
   }
 
   onCollision() {
-    this.score -= 500;
-    this.currentMovement = Movement.Stopped;
+    if (!this.submarine.flashing) {
+      this.submarine.moving = false
+      this.submarine.setVelocityX(0)
+      this.score -= 50;
+      this.submarine.startFlash()
+    }
   }
 
 
   dealWithKeyDown(event) {
 
+
     switch (event.which) {
       case Phaser.Input.Keyboard.KeyCodes.ONE:
         this.submarine.light.lightDown()
         break;
+      case Phaser.Input.Keyboard.KeyCodes.NUMPAD_ONE:
```

Comment:
```suggestion
      case Phaser.Input.Keyboard.KeyCodes.NUMPAD_ONE:
      case Phaser.Input.Keyboard.KeyCodes.ONE:
```

Tu peux mettre les deux cases a la suite pour mutualiser le code